src/serde/visitor.rs

```diff
@@ -204,7 +204,23 @@ impl<'a> de::Visitor<'a> for Visitor<Weekday> {
             6 => Ok(Weekday::Saturday),
             7 => Ok(Weekday::Sunday),
             _ => Err(E::invalid_value(
-                de::Unexpected::Unsigned(value.into()),
+                de::Unexpected::Signed(value),
+                &"a value in the range 1..=7",
+            )),
+        }
+    }
+
+    fn visit_u64<E: de::Error>(self, value: u64) -> Result<Weekday, E> {
```

Comment:
Both `visit_i64` and `visit_u64` are required to capture all integer types, from what I can tell? Excluding 128 bit types, but that's a separate story.

Reply:
Yes, it seems to be the case. In this commit I added `visit_u64` implementation for every `visit_i64` implementation.